diff --git a/internal/boxcli/root.go b/internal/boxcli/root.go
index 6e86a11..dd95ad9 100644
--- a/internal/boxcli/root.go
+++ b/internal/boxcli/root.go
@@ -11,6 +11,7 @@
 	"strings"
 	"time"
 
+	"github.com/carapace-sh/carapace"
 	"github.com/spf13/cobra"
 
 	"go.jetify.com/devbox/internal/boxcli/midcobra"
@@ -94,6 +95,7 @@ func RootCmd() *cobra.Command {
 	debugMiddleware.AttachToFlag(command.PersistentFlags(), "debug")
 	traceMiddleware.AttachToFlag(command.PersistentFlags(), "trace")
 
+	carapace.Gen(command)
 	return command
 }
 
