diff --git a/src/main.rs b/src/main.rs
index 68fea9c4..c433b989 100644
--- a/src/main.rs
+++ b/src/main.rs
@@ -11,6 +11,10 @@ use std::ffi::OsStr;
 use std::path::PathBuf;
 use tracing::{error, info};
 
+use carapace_spec_clap::Spec;
+use clap_complete::generate;
+use std::io;
+
 mod cmd;
 
 const VERSION: &str = concat!("v", clap::crate_version!());
@@ -18,7 +22,11 @@ const VERSION: &str = concat!("v", clap::crate_version!());
 fn main() {
     init_logger();
 
-    let command = create_clap_command();
+    let mut command = create_clap_command();
+    generate(Spec, &mut command, "mdbook", &mut io::stdout());
+    if true {
+        return;
+    }
 
     // Check which subcommand the user ran...
     let res = match command.get_matches().subcommand() {
