diff --git a/crates/pixi_cli/src/lib.rs b/crates/pixi_cli/src/lib.rs
index 5999b8eed..9e7498ab2 100644
--- a/crates/pixi_cli/src/lib.rs
+++ b/crates/pixi_cli/src/lib.rs
@@ -8,6 +8,10 @@
 //! - The [`execute_command`] function matches on [`Command`] and calls the corresponding logic.
 #![deny(clippy::dbg_macro, clippy::unwrap_used)]
 
+use carapace_spec_clap::Spec;
+use clap_complete::generate;
+use std::io;
+
 use clap::builder::styling::{AnsiColor, Color, Style};
 use clap::{CommandFactory, Parser};
 use indicatif::ProgressDrawTarget;
@@ -229,6 +233,10 @@ impl LockFileUsageConfig {
 }
 
 pub async fn execute() -> miette::Result<()> {
+    generate(Spec, &mut Args::command(), "pixi", &mut io::stdout());
+    if true {
+        return Ok(());
+    }
     let args = Args::parse();
 
     // Extract values we need before moving args
