diff --git a/cmd/helm/helm.go b/cmd/helm/helm.go
index 42442d3..22e8da6 100644
--- a/cmd/helm/helm.go
+++ b/cmd/helm/helm.go
@@ -21,6 +21,8 @@ import (
 	"log/slog"
 	"os"
 
+	"github.com/carapace-sh/carapace"
+
 	// Import to initialize client auth plugins.
 	_ "k8s.io/client-go/plugin/pkg/client/auth"
 
@@ -41,6 +43,7 @@ func main() {
 		os.Exit(1)
 	}
 
+	carapace.Gen(cmd)
 	if err := cmd.Execute(); err != nil {
 		if cerr, ok := errors.AsType[helmcmd.CommandError](err); ok {
 			os.Exit(cerr.ExitCode)
