diff --git a/main.go b/main.go
index cfcf1a8..8646892 100644
--- a/main.go
+++ b/main.go
@@ -22,6 +22,7 @@ import (
 	"github.com/caarlos0/env/v11"
 	"github.com/charmbracelet/log"
 	gap "github.com/muesli/go-app-paths"
+	"github.com/carapace-sh/carapace"
 	"github.com/spf13/cobra"
 	"github.com/spf13/viper"
 	"golang.org/x/term"
@@ -377,6 +378,7 @@ func main() {
 		fmt.Println(err)
 		os.Exit(1)
 	}
+	carapace.Gen(rootCmd)
 	if err := rootCmd.Execute(); err != nil {
 		_ = closer()
 		os.Exit(1)
