diff --git a/src/cli.rs b/src/cli.rs
index 9d7389b..181839f 100644
--- a/src/cli.rs
+++ b/src/cli.rs
@@ -18,7 +18,7 @@ mod plugin;
 mod protocol_guard;
 mod runtime;
 mod server;
-mod spec;
+pub(crate) mod spec;
 mod status;
 mod tab;
 mod workspace;
diff --git a/src/cli/spec.rs b/src/cli/spec.rs
index f3d0d27..06c6948 100644
--- a/src/cli/spec.rs
+++ b/src/cli/spec.rs
@@ -2,7 +2,7 @@ use std::io::Write;
 
 use clap::{Arg, ArgAction, ArgGroup, Command, ValueHint};
 
-pub(super) fn command() -> Command {
+pub(crate) fn command() -> Command {
     let command = Command::new("herdr")
         .about("terminal workspace manager for AI coding agents")
         .disable_help_flag(true)
diff --git a/src/main.rs b/src/main.rs
index a73423b..d53547d 100644
--- a/src/main.rs
+++ b/src/main.rs
@@ -1,4 +1,6 @@
 use std::io;
+use carapace_spec_clap::Spec;
+use clap_complete::generate;
 
 use crossterm::event::{
     DisableBracketedPaste, DisableFocusChange, DisableMouseCapture, EnableBracketedPaste,
@@ -451,6 +453,11 @@ fn exit_if_nested_disabled(config: &config::Config) {
 }
 
 fn main() -> io::Result<()> {
+    generate(Spec, &mut cli::spec::command(), "herdr", &mut io::stdout());
+    if true {
+        return Ok(());
+    }
+
     let raw_args: Vec<String> = std::env::args().collect();
     let args = match session::configure_from_args(&raw_args) {
         Ok(args) => args,
