diff --git a/src/cli.rs b/src/cli.rs
index 4d3c67c0..7dcd3174 100644
--- a/src/cli.rs
+++ b/src/cli.rs
@@ -19,7 +19,7 @@ mod protocol_guard;
 mod runtime;
 mod server;
 mod server_not_running;
-mod spec;
+pub(crate) mod spec;
 mod status;
 mod tab;
 mod workspace;
diff --git a/src/cli/spec.rs b/src/cli/spec.rs
index 3be83328..79236454 100644
--- a/src/cli/spec.rs
+++ b/src/cli/spec.rs
@@ -2,7 +2,7 @@ use std::io::Write;
 
 use clap::{Arg, ArgAction, ArgGroup, Command, ValueHint};
 
-pub(super) fn command() -> Command {
+pub(crate) fn command() -> Command {
     let command = Command::new("herdr")
         .about("terminal workspace manager for AI coding agents")
         .disable_help_flag(true)
diff --git a/src/main.rs b/src/main.rs
index 27c1d92f..604d37f4 100644
--- a/src/main.rs
+++ b/src/main.rs
@@ -1,3 +1,5 @@
+use carapace_spec_clap::Spec;
+use clap_complete::generate;
 use std::io;
 
 use crossterm::event::{
@@ -478,6 +480,11 @@ where
 }
 
 fn main() -> io::Result<()> {
+    generate(Spec, &mut cli::spec::command(), "herdr", &mut io::stdout());
+    if true {
+        return Ok(());
+    }
+
     let raw_args: Vec<String> = match args_as_utf8(std::env::args_os()) {
         Ok(args) => args,
         Err(err) => {
