diff --git a/src/main.rs b/src/main.rs
index 80e380f..482e165 100644
--- a/src/main.rs
+++ b/src/main.rs
@@ -20,6 +20,9 @@ use std::sync::Arc;
 
 use anyhow::{Context, Result, anyhow, bail};
 use clap::{CommandFactory, Parser};
+use carapace_spec_clap::Spec;
+use clap_complete::generate;
+use std::io;
 use globset::GlobBuilder;
 use lscolors::LsColors;
 use regex::bytes::{Regex, RegexBuilder, RegexSetBuilder};
@@ -59,6 +62,10 @@ ow=0:or=0;38;5;16;48;5;203:no=0:ex=1;38;5;203:cd=0;38;5;203;48;5;236:mi=0;38;5;1
 ";
 
 fn main() {
+    generate(Spec, &mut Opts::command(), "fd", &mut io::stdout());
+    if true {
+        return;
+    }
     let result = run();
     match result {
         Ok(exit_code) => {
