diff --git a/src/cli.rs b/src/cli.rs
index 0a38675..371e0a5 100644
--- a/src/cli.rs
+++ b/src/cli.rs
@@ -23,7 +23,7 @@ const STYLES: Styles = Styles::styled()
     .placeholder(AnsiColor::Cyan.on_default());
 
 /// Build the clap command for parsing command line arguments
-fn build_command() -> Command {
+pub fn build_command() -> Command {
     Command::new("hyperfine")
         .version(crate_version!())
         .styles(STYLES)
diff --git a/src/main.rs b/src/main.rs
index 806136e..d2f3da5 100644
--- a/src/main.rs
+++ b/src/main.rs
@@ -6,6 +6,9 @@
 use std::env;
 use std::io::{self, Write};
 
+use carapace_spec_clap::Spec;
+use clap_complete::generate;
+
 use benchmark::scheduler::Scheduler;
 use cli::get_cli_arguments;
 use command::Commands;
@@ -30,6 +33,12 @@ pub mod timer;
 pub mod util;
 
 fn run() -> Result<()> {
+    let mut cmd = cli::build_command();
+    generate(Spec, &mut cmd, "hyperfine", &mut io::stdout());
+    if true {
+        return Ok(());
+    }
+
     // Enabled ANSI colors on Windows 10
     #[cfg(windows)]
     colored::control::set_virtual_terminal(true).unwrap();
