diff --git a/cmd/glab/main.go b/cmd/glab/main.go
index fb47d0c55..466519c84 100644
--- a/cmd/glab/main.go
+++ b/cmd/glab/main.go
@@ -12,6 +12,7 @@ import (
 
 	"charm.land/fang/v2"
 	"charm.land/lipgloss/v2"
+	"github.com/carapace-sh/carapace"
 	"github.com/spf13/cobra"
 
 	"gitlab.com/gitlab-org/cli/internal/api"
@@ -113,6 +114,7 @@ func main() {
 		expandedArgs = os.Args[1:]
 	}
 	rootCmd := commands.NewCmdRoot(cmdFactory)
+	carapace.Gen(rootCmd)
 	cmd, _, err := rootCmd.Traverse(expandedArgs)
 
 	setupTelemetryHook(cfg, cmdFactory, cmd)
