diff --git a/internal/ghcmd/cmd.go b/internal/ghcmd/cmd.go
index 67b1564e2..1925f77a9 100644
--- a/internal/ghcmd/cmd.go
+++ b/internal/ghcmd/cmd.go
@@ -16,6 +16,7 @@ import (
 
 	surveyCore "github.com/AlecAivazis/survey/v2/core"
 	"github.com/AlecAivazis/survey/v2/terminal"
+	"github.com/carapace-sh/carapace"
 	"github.com/cli/cli/v2/api"
 	"github.com/cli/cli/v2/internal/agents"
 	"github.com/cli/cli/v2/internal/build"
@@ -172,6 +173,7 @@ func Main() exitCode {
 		fmt.Fprintf(stderr, "failed to create root command: %s\n", err)
 		return exitError
 	}
+	carapace.Gen(rootCmd)
 
 	expandedArgs := []string{}
 	if len(os.Args) > 0 {
