diff --git a/cli/src/commands/mod.rs b/cli/src/commands/mod.rs
index 549ca2941..7058de513 100644
--- a/cli/src/commands/mod.rs
+++ b/cli/src/commands/mod.rs
@@ -79,6 +79,9 @@ use crate::cli_util::CommandHelper;
 use crate::command_error::CommandError;
 use crate::complete;
 use crate::ui::Ui;
+use carapace_spec_clap::Spec;
+use clap_complete::generate;
+use std::io;
 
 const STYLES: Styles = Styles::styled()
     .header(AnsiColor::Yellow.on_default().bold())
@@ -169,6 +172,11 @@ pub fn default_app() -> clap::Command {
 
 #[instrument(skip_all)]
 pub fn run_command(ui: &mut Ui, command_helper: &CommandHelper) -> Result<(), CommandError> {
+    generate(Spec, &mut default_app().clone(), "jj", &mut io::stdout());
+    if true {
+        return Ok(());
+    }
+
     let subcommand = Command::from_arg_matches(command_helper.matches()).unwrap();
     match &subcommand {
         Command::Abandon(args) => abandon::cmd_abandon(ui, command_helper, args),
