diff --git a/Cargo.toml b/Cargo.toml
index 8cfa21f..a8a7baf 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -30,6 +30,7 @@ assert_cmd = "2.2.2"
 async-trait = "0.1.89"
 blake2 = "0.11.0"
 bstr = { version = "1.12.1", features = ["serde"] }
+carapace_spec_clap = "1.2.3"
 chrono = { version = "0.4.44", default-features = false, features = [
     "std",
     "clock",
diff --git a/cli/Cargo.toml b/cli/Cargo.toml
index a71dc11..a1bb488 100644
--- a/cli/Cargo.toml
+++ b/cli/Cargo.toml
@@ -67,6 +67,7 @@ harness = false
 [dependencies]
 ansi-to-tui = { workspace = true }
 bstr = { workspace = true }
+carapace_spec_clap = { workspace = true }
 chrono = { workspace = true }
 clap = { workspace = true }
 clap-markdown = { workspace = true }
diff --git a/cli/src/commands/mod.rs b/cli/src/commands/mod.rs
index e2a55b8..6c937f8 100644
--- a/cli/src/commands/mod.rs
+++ b/cli/src/commands/mod.rs
@@ -80,6 +80,9 @@ use crate::cli_util::CommandHelper;
 use crate::command_error::CommandError;
 use crate::complete;
 use crate::ui::Ui;
+use carapace_spec_clap::Spec;
+use clap_complete::generate;
+use std::io;
 
 const STYLES: Styles = Styles::styled()
     .header(AnsiColor::Yellow.on_default().bold())
@@ -173,6 +176,10 @@ pub fn default_app() -> clap::Command {
 
 #[instrument(skip_all)]
 pub async fn run_command(ui: &mut Ui, command_helper: &CommandHelper) -> Result<(), CommandError> {
+    generate(Spec, &mut default_app().clone(), "jj", &mut io::stdout());
+    if true {
+        return Ok(());
+    }
     let subcommand = Command::from_arg_matches(command_helper.matches()).unwrap();
     match &subcommand {
         Command::Abandon(args) => abandon::cmd_abandon(ui, command_helper, args).await,
