diff --git a/src/bin/cargo/cli.rs b/src/bin/cargo/cli.rs
index 7a16b5d48..719c0cfa2 100644
--- a/src/bin/cargo/cli.rs
+++ b/src/bin/cargo/cli.rs
@@ -18,13 +18,24 @@ use crate::util::is_rustup;
 use cargo::core::shell::ColorChoice;
 use cargo::util::style;
 
+use carapace_spec_clap::Spec;
+use clap::{Arg, ArgAction, Command};
+use clap_complete::generate;
+use std::io;
+
 #[tracing::instrument(skip_all)]
 pub fn main(gctx: &mut GlobalContext) -> CliResult {
     // CAUTION: Be careful with using `config` until it is configured below.
     // In general, try to avoid loading config values unless necessary (like
     // the [alias] table).
 
-    let args = cli(gctx).try_get_matches()?;
+    let mut cmd = cli(gctx);
+    generate(Spec, &mut cmd, "cargo", &mut io::stdout());
+    if true {
+        return Ok(());
+    }
+
+    let args = cmd.try_get_matches()?;
 
     // Update the process-level notion of cwd
     if let Some(new_cwd) = args.get_one::<std::path::PathBuf>("directory") {
