diff --git a/cmd/minikube/cmd/root.go b/cmd/minikube/cmd/root.go
index 0d35165..87318e9 100644
--- a/cmd/minikube/cmd/root.go
+++ b/cmd/minikube/cmd/root.go
@@ -24,6 +24,7 @@ import (
 	"runtime"
 	"strings"
 
+	"github.com/carapace-sh/carapace"
 	"github.com/spf13/cobra"
 	"github.com/spf13/pflag"
 	"github.com/spf13/viper"
@@ -171,6 +172,7 @@ func Execute() {
 		klog.Errorf("oci env: %v", err)
 	}
 
+	carapace.Gen(RootCmd)
 	if err := RootCmd.Execute(); err != nil {
 		// Cobra already outputs the error, typically because the user provided an unknown command.
 		defer os.Exit(reason.ExProgramUsage)
