diff --git a/cmd/minikube/cmd/root.go b/cmd/minikube/cmd/root.go
index 466ffbe..7e76d81 100644
--- a/cmd/minikube/cmd/root.go
+++ b/cmd/minikube/cmd/root.go
@@ -25,6 +25,7 @@ import (
 	"slices"
 	"strings"
 
+	"github.com/carapace-sh/carapace"
 	"github.com/spf13/cobra"
 	"github.com/spf13/pflag"
 	"github.com/spf13/viper"
@@ -165,6 +166,7 @@ func Execute() {
 		klog.Errorf("oci env: %v", err)
 	}
 
+	carapace.Gen(RootCmd)
 	if err := RootCmd.Execute(); err != nil {
 		// Cobra already outputs the error, typically because the user provided an unknown command.
 		defer os.Exit(reason.ExProgramUsage)
