diff --git a/cmd/root.go b/cmd/root.go
index 8e9448f7..f0314e68 100644
--- a/cmd/root.go
+++ b/cmd/root.go
@@ -12,6 +12,7 @@ import (
 
 	goversion "github.com/caarlos0/go-version"
 	"github.com/caarlos0/log"
+	"github.com/carapace-sh/carapace"
 	"github.com/charmbracelet/fang"
 	"github.com/charmbracelet/lipgloss/v2"
 	"github.com/goreleaser/goreleaser/v2/internal/gerrors"
@@ -127,6 +128,7 @@ goreleaser release
 		newMcpCmd(version).cmd,
 	)
 	root.cmd = cmd
+	carapace.Gen(cmd)
 	return root
 }
 
