diff --git a/src/main.rs b/src/main.rs
index 2d3fffc..6fbd42b 100644
--- a/src/main.rs
+++ b/src/main.rs
@@ -1,4 +1,8 @@
 #![allow(clippy::needless_return)]
+use clap::CommandFactory;
+use carapace_spec_clap::Spec;
+use clap_complete::generate as clap_generate;
+use std::io;
 
 use ripsecrets::find_secrets;
 use std::process;
@@ -23,6 +27,10 @@ enum RunResult {
 }
 
 fn main() {
+    clap_generate(Spec, &mut Args::command(), "ripsecrets", &mut io::stdout());
+    if true {
+        return;
+    }
     match run() {
         RunResult::PreCommitInstallSuccessful => process::exit(0),
         RunResult::NoSecretsFound => process::exit(0),
