diff --git a/gum.go b/gum.go
index 3b9d1a4..db7e7f7 100644
--- a/gum.go
+++ b/gum.go
@@ -225,4 +225,6 @@ type Gum struct {
 	// $ gum version-check '~> 0.15'
 	//
 	VersionCheck version.Options `cmd:"" help:"Semver check current gum version"`
+
+	kong.Plugins
 }
diff --git a/main.go b/main.go
index 38b8a62..a5c6b70 100644
--- a/main.go
+++ b/main.go
@@ -11,6 +11,7 @@ import (
 	"charm.land/gum/v2/internal/exit"
 	"charm.land/lipgloss/v2"
 	"github.com/alecthomas/kong"
+	"github.com/carapace-sh/carapace-spec-kong"
 )
 
 const shaLen = 7
@@ -43,6 +44,7 @@ func main() {
 	}
 
 	gum := &Gum{}
+	gum.Plugins = kong.Plugins{&spec.Plugin{}}
 	ctx := kong.Parse(
 		gum,
 		kong.Description(fmt.Sprintf("A tool for %s shell scripts.", bubbleGumPink.Render("glamorous"))),
