diff --git a/src/cli/rustup_mode.rs b/src/cli/rustup_mode.rs
index d1b43882..a70b7670 100644
--- a/src/cli/rustup_mode.rs
+++ b/src/cli/rustup_mode.rs
@@ -56,6 +56,9 @@ use crate::{
     utils::{self, ExitCode},
 };
 
+use carapace_spec_clap::Spec;
+use clap_complete::generate;
+
 const TOOLCHAIN_OVERRIDE_ERROR: &str = "To override the toolchain using the 'rustup +toolchain' syntax, \
                         make sure to prefix the toolchain override with a '+'";
 
@@ -597,6 +600,11 @@ pub async fn main(
 ) -> Result<ExitCode> {
     self_update::cleanup_self_updater(process)?;
 
+    generate(Spec, &mut Rustup::command(), "rustup", &mut io::stdout());
+    if true {
+        return Ok(utils::ExitCode(0));
+    }
+
     use clap::error::ErrorKind::*;
     let matches = match Rustup::try_parse_from(process.args_os()) {
         Ok(matches) => matches,
